data/js/item/bagofsending.js

```diff
@@ -0,0 +1,177 @@
+{
+	const maxCharges = 30;
+	if( objType == 0 ) 
+	{
+		var Charges = 0;
+		switch( RandomNumber( 0, 2 ))
+		{
+			case 0: Charges = 3; break;
+			case 1: Charges = 6; break;
+			case 2: Charges = 9; break;
+		}
+		objMade.SetTag( "charges", Charges )
+		objMade.SetTag( "maxCharges", maxCharges )
+	}
+}
+
+function onUseChecked( pUser, iUsed )
+{
+	var socket = pUser.socket;
+	var Charges = iUsed.GetTag( "charges" )
+	if( pUser.visible == 1 || pUser.visible == 2 )
+	{
+		pUser.visible = 0;
+	}
+
+	if( socket && iUsed && iUsed.isItem )
+	{
+		//Check to see if it's locked down
+		if( iUsed.movable == 2 || iUsed.movable == 3 )
+		{
+			socket.SysMessage( GetDictionaryEntry( 774, socket.language )); // That is locked down and you cannot use it
+			return false;
+		}
+
+		var itemOwner = GetPackOwner( iUsed, 0 );
+		if( itemOwner == null || itemOwner.serial != pUser.serial ) 
+		{
+			socket.SysMessage( GetDictionaryEntry( 19070, socket.language )); // The bag of sending must be in your backpack.
+			return false;
+		}
+		else if( Charges == 0 )
+		{
+			socket.SysMessage( GetDictionaryEntry( 0, socket.language)); // This item is out of charges.
+			return false;
+		}
+		else
+		{
+			socket.tempObj = iUsed;
+			socket.CustomTarget( 0 );
+			iUsed.Refresh();
+		}
+	}
+	return false;
+}
+
+function onCallback0( socket, myTarget)
+{
+	var pUser = socket.currentChar;
+	var requiredCharges = 1;
+	var iUsed = socket.tempObj;
+	var Charges = parseInt( iUsed.GetTag( "charges" ));
+	if( pUser.visible == 1 || pUser.visible == 2 )
+	{
+		pUser.visible = 0;
+	}
+
+	var itemOwner = GetPackOwner( iUsed, 0 );
+	if( itemOwner == null || itemOwner.serial != pUser.serial )
+	{
+		pUser.SysMessage( GetDictionaryEntry( 19070, socket.language )); // The bag of sending must be in your backpack
+		return false;
+	}
+	else if( Charges == 0 )
+	{
+		pUser.SysMessage( GetDictionaryEntry( 0, socket.language )); // This item is out of charges.
+		return false;
+	}
+
+	if( !socket.GetWord( 1 ) && myTarget.isItem )
+	{
+		var bankBox = pUser.FindItemLayer( 29 );
+		if( !ValidateObject( bankBox ))
+		{
+			socket.SysMessage( GetDictionaryEntry( 19071, socket.language ));// Error: No valid bankbox found! Please contact a GM for assistance.
+			return false;
+		}
+		else if( !ValidateObject( myTarget.container ) || myTarget.container != pUser.pack )
+		{
+			socket.SysMessage( GetDictionaryEntry( 19072, socket.language )); // You may only send items from your backpack to your bank box.
+			return false;
+		}
+		else if( myTarget.type == 1 || myTarget == iUsed )
+		{
+			socket.SysMessage( GetDictionaryEntry( 19073, socket.language )); // You may only send items from your backpack to your bank box.
+			return false;
+		}
+		else if( bankBox.totalItemCount >= bankBox.maxItems )
+		{
+			socket.SysMessage( GetDictionaryEntry( 19074, socket.language )); // Your bank box is full.
+			return false;
+		}
+		else if( bankBox.weight > bankBox.weightMax )
```

Comment:
One thing this does not check, is whether the player's bankbox will be overloaded AFTER the item has been sent to the bank. This should also check if bankBox.weight + ( myTarget.weight * myTarget.amount ) > bankBox.weightMax